tests_backend/tests/cli/test_cli.ts

```diff
@@ -230,6 +247,111 @@ async function testSelectableDisallow(_backendBaseUrl: URL, supabase: SupabaseTy
   }
 }
 
+function increaseSemver() {
+  const lastNumber = Number.parseInt(semver.charAt(semver.length - 1))
+  const newSemver = `${semver.slice(0, -1)}${(lastNumber + 1).toString()}`
+  semver = newSemver
+}
+
+async function testCompatibilityTable(_backendBaseUrl: URL, _supabase: SupabaseType) {
+  dependencies = {
+    '@capacitor/android': '^4.5.0',
+  }
+  await Deno.writeTextFile(`${tempFileFolder}/package.json`, defaultPackageJson.replace('%DEPENDENCIES%', JSON.stringify(dependencies)))
+
+  await pnpmInstall()
+
+  async function assertCompatibilityTableColumns(column1: string, column2: string, column3: string, column4: string) {
+    const cliTableOutput = await runCli(['bundle', 'compatibility', '-c', 'production'])
+    const androidPackage = cliTableOutput.split('\n').find(l => l.includes('@capacitor/android'))
+
+    assert(androidPackage !== undefined, 'Android package is not found in compatibility table')
+    const androidPackageSplit = androidPackage!.split('│').slice(2, -1)
+    assert(androidPackageSplit.length === 4, `Android package does not have 4 columns (It has ${androidPackageSplit.length} columns)`)
+
+    assert(androidPackageSplit[0].includes(column1), `Android package name is not ${column1} (It is ${androidPackageSplit[0]})`)
+    assert(androidPackageSplit[1].includes(column2), `Android local package version is not ${column2} (It is ${androidPackageSplit[1]})`)
+    assert(androidPackageSplit[2].includes(column3), `Android remote package version is not ${column3} (It is ${androidPackageSplit[2]})`)
+    assert(androidPackageSplit[3].includes(column4), `Android compatible is not a ${column4} (It is ${androidPackageSplit[3]})`)
+  }
+
+  await assertCompatibilityTableColumns('@capacitor/android', '4.5.0', 'None', '❌')
+
+  // Let's upload now a new version
+  increaseSemver()
+
+  // Re run the upload
+  const uploadCli = await runCli(['bundle', 'upload', '-b', semver, '-c', 'production'])
+  assert(uploadCli.includes('Time to share your update to the world'), `CLI output does not include 'Time to share your update to the world'. CLI output:\n${uploadCli}`)
+
+  // Let's re run the compatibility table
+  await assertCompatibilityTableColumns('@capacitor/android', '4.5.0', '4.5.0', '✅')
+
+  // Now let's remove the package and run the compatibility table again
+  dependencies = {}
+  await Deno.writeTextFile(`${tempFileFolder}/package.json`, defaultPackageJson.replace('%DEPENDENCIES%', JSON.stringify(dependencies)))
+
+  await assertCompatibilityTableColumns('@capacitor/android', 'None', '4.5.0', '❌')
+
+  await pnpmInstall()
+}
+
+async function testAutoMinVersionFlag(_backendBaseUrl: URL, supabase: SupabaseType) {
+  // At this stage the lastest upload has the `@capacitor/android`. We do NOT have this package installed thus the new upload will not be compatible
+  // Let's upload now a new version and check if this statement is correct
+
+  async function uploadWithAutoFlag(expected: string): Promise<string> {
```

Comment:
Can we change the method name? It is not clear that it is asserting anything, judging by the name, looks like it'll upload with the auto flag and that's it

Reply:
```suggestion
  async function uploadWithAutoFlagAndAssertMinVersion(expected: string): Promise<string> {
```

Reply:
```suggestion
  async function assertMinVersionAfterUploadWithAutoFlag(expected: string): Promise<string> {
```

Reply:
left two examples but feel free to change as you prefer

Reply:
Resolved [here](https://github.com/Cap-go/capgo/pull/419/commits/a3edb16729610d0889b7825e36a10cac3e41a15e)